sunscreen_math/src/poly/mod.rs

```diff
@@ -75,15 +252,75 @@ where
     }
 }
 
+#[refify_binary_op]
+impl<T> Mul<&T> for &Polynomial<T>
+where
+    T: Ring,
+{
+    type Output = Polynomial<T>;
+
+    fn mul(self, rhs: &T) -> Self::Output {
+        Self::Output {
+            coeffs: self
+                .coeffs
+                .iter()
+                .map(|x| x.clone() * rhs)
```

Comment:
Huh... this doesn't look right to me :thinking:  Why component wise coefficient multiplication here?

Reply:
I'm guessing this is made possible by the Barret backend. If so, we should perhaps document that correctness depends on choices of `T`, or better yet have a trait constraint that enforces it, as I don't think every ring would work here. But I am brain dead so maybe I'm missing something.

Reply:
Oh my bad - I see, this isn't component-wise

Reply:
This has nothing to do with the Barrett backend and the `Ring` doesn't imply Barrett reductions. Coefficient-wise multiplication is sensible on polynomials in NTT or FFT form, but that's completely unrelated. Look again and you'll see we're multiplying each coefficient in lhs by `rhs`, not component-wise multiplication. This is simple scalar multiplication.

Take for example, `Z[X] x Z`. `a in Z` is equivalent to the polynomial `a*x^0`, so it's well defined.